x/clp/types/keys.go

```diff
@@ -0,0 +1,83 @@
+package types
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	// ModuleName is the name of the module
+	ModuleName = "clp"
+
+	// StoreKey to be used when creating the KVStore
+	StoreKey = ModuleName
+
+	// RouterKey to be used for routing msgs
+	RouterKey = ModuleName
+
+	// QuerierRoute to be used for querier msgs
+	QuerierRoute = ModuleName
+
+	NativeTicker = "rwn"
+	NativeChain  = "SIFCHAIN"
+	NativeSymbol = "RWN"
+)
+
+var (
+	PoolPrefix              = []byte{0x00} // key for storing Pools
+	LiquidityProviderPrefix = []byte{0x01} // key for storing Liquidity Providers
+)
+
+func GetPoolKey(externalTicker string, nativeTicker string) ([]byte, error) {
+	addr, err := GetPoolAddress(externalTicker, nativeTicker)
+	if err != nil {
+		return nil, err
+	}
+	key := []byte(addr)
+	return append(PoolPrefix, key...), nil
+}
+
+//Generate a new pool address from a string
+//The external asset ticker and the native asset ticket ,in combination is used to generate an unique address
+func GetPoolAddress(externalTicker string, nativeTicker string) (string, error) {
+	addr, err := GetAddress(externalTicker, nativeTicker)
+	if err != nil {
+		return "", err
+	}
+	return addr.String(), nil
+}
+
+func GetAddress(externalTicker, nativeTicker string) (sdk.AccAddress, error) {
```

Comment:
@AustinoBombino , would be a better person to answer this

Reply:
Yeah at the time of writing the architecture I expected this to evolve and clarify through development but I overlooked reassessing it. 

But I think you're right a single address in the form of a module account would be better. It would simplify a lot. I believe the supply module has been merged into the bank module but regardless it would make things more readable and eliminate the need for pool to pool transfers during double swaps.

Reply:
Hey @AustinoBombino can you create a asana task for this and spec it out a bit. I want to get this PR merged and this change can up as a future improvement before we hit mainnet.

Reply:
created task here: https://app.asana.com/0/1196711589172801/1198992855386359/f